LoRaEngine/LoraKeysManagerFacade/LoRaADRFunction/ADRExecutionItem.cs

```diff
@@ -19,7 +19,7 @@ public ADRExecutionItem(ILoRaADRManager adrManager)
         public async Task<FunctionBundlerExecutionState> ExecuteAsync(IPipelineExecutionContext context)
         {
             context.Result.AdrResult = await this.HandleADRRequest(context.DevEUI, context.Request.AdrRequest);
-            context.Result.NextFCntDown = context.Result?.AdrResult.FCntDown > 0 ? context.Result.AdrResult.FCntDown : (uint?)null;
+            context.Result.NextFCntDown = context.Result.AdrResult != null && context.Result.AdrResult.FCntDown > 0 ? context.Result.AdrResult.FCntDown : (uint?)null;
```

Comment:
Isn't that similar?

Reply:
Don't think so, right? The first one was testing if Result was null and the new one does validate AdrResult.

Reply:
PSA: I shot myself in the foot with a bug using nullables. They are dangerous for == and != comparison.
```csharp
int? a = null;
if (a == 0) {
    // will not go in since null != 0
}

if (a != 0) {
    // will go in, since 0 != null
}
```

What you probably want is:

```csharp
if ((a ?? 0) == 0) {
    // will go in if a == null || a == 0
}

if ((a ?? 0) != 0) {
    // will go in only if 'a' has a value and is not zero
}
```

Reply:
So `.Result` can never be null, right?

Reply:
Correct I missed a ? :P

Reply:
@fbeltrao Correct, .Result can never be null. It has a property initializer. The null ref came from AdrResult. Here the FCntDown (on the AdrResult) is not a nullable. Should all be ok here.